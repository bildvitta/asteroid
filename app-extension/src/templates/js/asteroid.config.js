module.exports = {
  /**
   * Configurações de API
   * @type {{ serverTimeout: number=10000 }}
  */
  api: {
    serverTimeout: 10000
  },

  framework: {
    /**
     * O asteroid ao identificar que a aplicação tenha dependências terceiras e
     * não estão declarados na opção "thirdPartyComponents" ele remove as dependências da aplicação (package.json), caso precise dessas dependências mesmo que sem os thirdPartyComponents, desative esta opção setando como "false".
     * @type {boolean=true}
    */
    autoRemoveThirdDependencies: true,

    fonts: {
      observer: {
        /**
         * O asteroid adiciona um observer nas fontes, enquanto ela não carrega adiciona um loading para
         * não quebrar o layout, a configuração "waitForUserAuthenticate: true" define que o observer deve esperar
         * o usuário estar autenticado para começar a observar as fontes, para isto a aplicação deve poder receber um
         * "postMessage" com um evento type "requestUser" e responder com um evento type "responseUser".
         *
         * @type {boolean=true}
         * @example
         * // solicita o usuário
         * window.postMessage({ type: 'requestUser' })
         *
         * window.addEventListener('message', ({ data }) => data.type // responseUser })
       */
        waitForUserAuthenticate: true
      }
    },

    /**
     * Controla o sistema de notificações da aplicação, ao ativado e configurado,
     * estará disponível os seguintes recursos:
     *
     * - Item de notificação no componente QasAppUser;
     * - Toast de notificação em real time;
     * - Ícone de notificação no menu em real time.
     *
     * @type {{ useNotifications: boolean }}
     */
    featureToggle: {
      useNotifications: false
    },

    /**
     * Componentes que necessitam de instalação de bibliotecas de terceiros
     * dentro da aplicação que utiliza o asteroid, a instalação é feita automaticamente pelo asteroid.
     * @type {Array<'QasChartView' | 'QasMap'>}
    */
    thirdPartyComponents: []
  }
}
