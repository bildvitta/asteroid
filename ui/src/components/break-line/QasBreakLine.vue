<template>
  <component :is="parentTag">
    <component :is="tag" v-for="(item, index) in items" :key="index" :class="tagClass" :style="tagStyle">
      {{ item }}
    </component>
  </component>
</template>

<script>
export default {
  props: {
    parentTag: {
      default: 'div',
      type: String
    },

    tag: {
      default: 'div',
      type: String
    },

    tagClass: {
      default: null,
      type: [Array, Object, String]
    },

    tagStyle: {
      default: null,
      type: [Array, Object, String]
    },

    text: {
      default: '',
      type: String
    },

    split: {
      default: '\n',
      type: String
    }
  },

  computed: {
    items () {
      const slot = this.$slots.default
      const text = this.text || (slot ? slot[0].text : '')

      return text.split(this.split)
    }
  }
}
</script>
