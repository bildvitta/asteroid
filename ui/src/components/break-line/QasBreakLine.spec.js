import { mount } from '@vue/test-utils'

import '@/index.js'
import QasBreakLine from './QasBreakLine.vue'

describe('Test QasBreakLine component', () => {
  const wrapper = mount(QasBreakLine, {
    propsData: {
      text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. \n Error tempora vero aspernatur facere. Deserunt delectus amet eveniet velit reprehenderit itaque',
      tagClass: 'my-class'
    }
  })

  it('test defaults', () => {
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.findAll('div')).toHaveLength(3)
    expect(wrapper.findAll('.my-class')).toHaveLength(2)
    expect(wrapper.isVisible()).toBe(true)
  })

  it('change parentTag', async () => {
    await wrapper.setProps({ parentTag: 'section' })

    expect(wrapper.find('section').isVisible()).toBe(true)
  })

  it('change tag', async () => {
    await wrapper.setProps({ tag: 'span' })

    expect(wrapper.findAll('span')).toHaveLength(2)
  })
})
