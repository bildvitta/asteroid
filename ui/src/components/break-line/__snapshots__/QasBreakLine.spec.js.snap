// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test QasBreakLine component test defaults 1`] = `
<div>
  <div
    class="my-class"
  >
    
    Lorem ipsum dolor sit amet consectetur adipisicing elit. 
  
  </div>
  <div
    class="my-class"
  >
    
     Error tempora vero aspernatur facere. Deserunt delectus amet eveniet velit reprehenderit itaque
  
  </div>
</div>
`;
